libs/usb-drive/src/usb_drive.ts

```diff
@@ -0,0 +1,143 @@
+import { assertDefined } from '@votingworks/basics';
+import makeDebug from 'debug';
+
+const exec = promisify(execFile);
+
+const debug = makeDebug('usb-drive');
+
+export type UsbDriveStatus =
+  | { status: 'no_drive' }
+  | { status: 'mounted'; mountPoint: string }
+  | { status: 'ejected' }
+  | { status: 'error'; reason: 'bad_format' };
+
+export interface UsbDrive {
+  status(): Promise<UsbDriveStatus>;
+  eject(): Promise<void>;
+  format(): Promise<void>;
+}
+
+interface BlockDeviceInfo {
+  name: string;
+  path: string;
+  mountpoint: string | null;
+  fstype: string | null;
+  fsver: string | null;
+  label: string | null;
+}
+
+async function getBlockDeviceInfo(
+  devicePath: string
+): Promise<BlockDeviceInfo | undefined> {
+  try {
+    const { stdout } = await exec('lsblk', [
+      '-J',
+      '-n',
+      '-l',
+      '-o',
+      ['NAME', 'MOUNTPOINT', 'FSTYPE', 'FSVER', 'LABEL'].join(','),
+      devicePath,
+    ]);
+    const rawData = JSON.parse(stdout) as {
+      blockdevices: BlockDeviceInfo[];
+    };
+    debug(`Got block device info for ${devicePath}: ${stdout}`);
+    return { ...assertDefined(rawData.blockdevices[0]), path: devicePath };
+  } catch (error) {
+    debug(`Error getting block device info for ${devicePath}: ${error}`);
+    return undefined;
+  }
+}
+
+async function findUsbDriveDevice(): Promise<string | undefined> {
+  const DEVICE_PATH_PREFIX = '/dev/disk/by-id/';
+  const USB_REGEXP = /^usb(.+)part(.*)$/;
+
+  // List devices, filtering to only the USB partitions
+  const devicesById = (await fs.readdir(DEVICE_PATH_PREFIX)).filter((name) =>
+    USB_REGEXP.test(name)
+  );
+
+  // We only support one USB drive at a time
+  const [deviceId] = devicesById;
+  if (!deviceId) {
+    return undefined;
+  }
+
+  // Follow the symlink
+  const devicePath = join(
+    DEVICE_PATH_PREFIX,
+    await fs.readlink(join(DEVICE_PATH_PREFIX, deviceId))
+  );
+
+  return devicePath;
+}
+
+async function getUsbDriveStatus(): Promise<BlockDeviceInfo | undefined> {
+  const devicePath = await findUsbDriveDevice();
+  if (!devicePath) return undefined;
+  debug(`Found USB drive at ${devicePath}`);
+  return await getBlockDeviceInfo(devicePath);
+}
+
+const VX_MOUNT_POINT = '/media/vx/usb-drive';
+
+async function mountUsbDrive(devicePath: string): Promise<void> {
+  debug(`Mounting USB drive ${devicePath} at ${VX_MOUNT_POINT}`);
+  await fs.mkdir(VX_MOUNT_POINT, { recursive: true });
+  await exec('mount', [
+    '-w',
+    '-o',
+    'umask=000,nosuid,nodev,noexec',
+    devicePath,
+    VX_MOUNT_POINT,
+  ]);
+}
+
+async function unmountUsbDrive(mountPoint: string): Promise<void> {
+  debug(`Unmounting USB drive at ${mountPoint}`);
+  await exec('umount', [mountPoint]);
+}
+
+export function detectUsbDrive(): UsbDrive {
+  let didEject = false;
+
+  return {
+    async status(): Promise<UsbDriveStatus> {
+      let deviceInfo = await getUsbDriveStatus();
+      if (!deviceInfo) return { status: 'no_drive' };
```

Comment:
I'm guessing we'd `didEject = false` here

Reply:
I don't totally understand what you're getting at here. Can you elaborate?

Reply:
I don't see anywhere where we clear a `didEject = true` status once the USB drive is no longer inserted - am I missing it or is it missing?

Reply:
🤦 great catch. Fixed and added regression test: 6d174455b